File: app_canibuythat/src/main/java/com/gb/canibuythat/api/model/ApiTransaction.java
package com.gb.canibuythat.api.model;

import android.support.annotation.Keep;

import com.google.gson.annotations.SerializedName;

import lombok.Getter;

@Keep
@Getter
public class ApiTransaction {

    @SerializedName("account_balance")
    private int accountBalance;

    @SerializedName("amount")
    private int amount;

    @SerializedName("created")
    private String created;

    @SerializedName("currency")
    private String currency;

    @SerializedName("description")
    private String description;

    @SerializedName("id")
    private String id;

    @SerializedName("merchant")
    private String merchant;

    @SerializedName("notes")
    private String notes;

    @SerializedName("is_load")
    private boolean isLoad;

    @SerializedName("settled")
    private String settled;

    @SerializedName("category")
    private String category;
}

File: app_canibuythat/src/main/java/com/gb/canibuythat/api/model/ApiTransaction.kt
package com.gb.canibuythat.api.model

class ApiTransaction(
        val amount: Int,
        val created: String,
        val description: String,
        val id: String,
        val notes: String,
        val category: String,
        val include_in_spending: Boolean,
        val is_load: Boolean) {
    override fun toString(): String {
        return "ApiTransaction(category='$category', amount=$amount, created='$created', description='$description', notes='$notes', include_in_spending=$include_in_spending, id='$id', is_load=$is_load)"
    }
}